src/compiler/transformers/classFields.ts

```diff
@@ -218,19 +348,47 @@ namespace ts {
         }
 
         function createPrivateIdentifierAccess(info: PrivateIdentifierInfo, receiver: Expression): Expression {
-            receiver = visitNode(receiver, visitor, isExpression);
-            switch (info.placement) {
-                case PrivateIdentifierPlacement.InstanceField:
-                    return context.getEmitHelperFactory().createClassPrivateFieldGetHelper(
-                        nodeIsSynthesized(receiver) ? receiver : factory.cloneNode(receiver),
-                        info.weakMapName
+            return createPrivateIdentifierAccessHelper(info, visitNode(receiver, visitor, isExpression));
+        }
+
+        function createPrivateIdentifierAccessHelper(info: PrivateIdentifierInfo, receiver: Expression): Expression {
+            let getExpression: Expression;
+
+            switch(info.kind) {
+                case PrivateIdentifierKind.Accessor:
+                    getExpression = context.getEmitHelperFactory().createClassPrivateFieldGetHelper(
+                        receiver,
+                        info.brandCheckIdentifier,
+                        info.kind,
+                        info.getterName
                     );
-                default: return Debug.fail("Unexpected private identifier placement");
+                    break;
+                case PrivateIdentifierKind.Method:
+                    getExpression = context.getEmitHelperFactory().createClassPrivateFieldGetHelper(
+                        receiver,
+                        info.brandCheckIdentifier,
+                        info.kind,
+                        info.methodName
+                    );
+                    break;
+                case PrivateIdentifierKind.Field:
+                    getExpression = context.getEmitHelperFactory().createClassPrivateFieldGetHelper(
+                        receiver,
+                        info.brandCheckIdentifier,
+                        info.kind,
+                        info.variableName
+                    );
+                    break;
+                default:
+                    Debug.assertNever(info, "Unknown private element type");
             }
+
+            setCommentRange(receiver, getExpression);

```

Comment:
What this is doing is setting the comment range of `receiver` to `{ pos: -1, end: -1 }` (from `getExpression`). What you actually want to do is set the comment range for the entire helper to the property access it replaces.

So, for example:

**Property Access:**
```ts
   this.#x
// |      |
// |      +-----------------------------+
// | (pos)                              | (end)
// v                                    v
   __classPrivateFieldGet(this, _x, "f")
```

**Property Assignment:**
```ts
   this.#x = 1
// |          |
// |          +----------------------------+
// | (pos)                                 | (end)
// v                                       v
   __classPrivateFieldSet(this, _x, 1, "f")
```

**Method Calls:**
```ts
   // range for the PropertyAccessExpression
   this.#x()
// |      |
// |      +-----------------------------------------+
// | (pos)                                          | (end)
// v                                                v
   __classPrivateFieldGet(this, _instances, "m", _x).call(this)

   // range for the CallExpression
   this.#x()
// |        |
// |        +--------------------------------------------------+
// | (pos)                                                     | (end)
// v                                                           v
   __classPrivateFieldGet(this, _instances, "m", _x).call(this)
```

You could, for example, add a `location: TextRange` argument to this function and call `setCommentRange(getExpression, location)` here, and then pass in the appropriate node to use for the comment range. You could also do `setSourceMapRange`, since I imagine the source maps aren't quite correct here either.

Reply:
I think this might be the right thing to do but it doesn't solve the duplicated comments that come from the `receiver`. You can see the changes to fixtures here: https://github.com/bloomberg/TypeScript/pull/65/files.

Also just want to confirm, we call `setOriginalNode` for the helper expressions with the original property access/assignment. Do we still need to call `setCommentRange`?

Reply:
`setOriginalNode` does not set `pos` and `end`, but will copy any existing *custom* comment and source map ranges. 

I stepped through this in the debugger and found the culprit. `emitList` in *emitter.ts* tries to emit intervening comments between any previous node and the current node. It does this by emitting the *trailing* comments of the start position of the node to catch any extra comments that may have been missed. 

A workaround is to use `setCommentRange(receiver, moveRangePos(receiver, -1))` so that the receiver's leading comment range is empty, and use something like `setTextRange` on the result of the call to `createPrivateIdentifierAccessHelper` to point it to the property access node. That way, leading comments will be preserved as part of the statement list, but will be ignored since the receiver is in an argument list.